File: backend/src/main/java/com/antshorttv/video/VideoDecompositionController.java
package com.antshorttv.video;

import com.antshorttv.common.ApiResponse;
import com.antshorttv.common.TenantRequestSupport;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;
import java.util.List;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequestMapping("/api/video-script-decomposition")
public class VideoDecompositionController {
    private final VideoDecompositionService service;

    public VideoDecompositionController(VideoDecompositionService service) {
        this.service = service;
    }

    @GetMapping("/batches")
    public ApiResponse<List<VideoDecompositionBatchResponse>> list(
        @RequestParam(required = false) Long projectId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.list(tenantId(request), projectId));
    }

    @PostMapping("/batches")
    public ApiResponse<VideoDecompositionBatchResponse> create(
        @Valid @RequestBody CreateVideoDecompositionBatchRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.create(tenantId(request), body, request));
    }

    @PostMapping("/uploads")
    public ApiResponse<VideoDecompositionUploadResponse> upload(
        @RequestParam Long projectId,
        @RequestParam("file") MultipartFile file,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.upload(tenantId(request), projectId, file));
    }

    @GetMapping("/batches/{batchId}")
    public ApiResponse<VideoDecompositionBatchResponse> detail(
        @PathVariable Long batchId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.detail(tenantId(request), batchId));
    }

    @GetMapping("/episodes/{episodeId}")
    public ApiResponse<VideoDecompositionEpisodeDetailResponse> episode(
        @PathVariable Long episodeId,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.episodeDetail(tenantId(request), episodeId));
    }

    @PostMapping("/episodes/{episodeId}/retry")
    public ApiResponse<VideoDecompositionEpisodeResponse> retry(
        @PathVariable Long episodeId,
        @RequestBody(required = false) RetryVideoDecompositionEpisodeRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.retry(tenantId(request), episodeId, body));
    }

    @PutMapping("/episodes/{episodeId}/draft")
    public ApiResponse<VideoDecompositionEpisodeResponse> updateDraft(
        @PathVariable Long episodeId,
        @Valid @RequestBody UpdateVideoDecompositionDraftRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.updateDraft(tenantId(request), episodeId, body));
    }

    @PostMapping("/episodes/{episodeId}/confirm")
    public ApiResponse<VideoDecompositionEpisodeResponse> confirm(
        @PathVariable Long episodeId,
        @Valid @RequestBody ConfirmVideoDecompositionDraftRequest body,
        HttpServletRequest request
    ) {
        return ApiResponse.success(service.confirmDraft(tenantId(request), episodeId, body));
    }

    private Long tenantId(HttpServletRequest request) {
        return TenantRequestSupport.tenantId(request);
    }
}

File: frontend/src/services/ant-design-pro/videoDecompositionController.ts
// @ts-ignore
/* eslint-disable */
import { request } from "@umijs/max";

/** 此处后端没有提供注释 GET /api/video-script-decomposition/batches */
export async function list(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.listParams,
  options?: { [key: string]: any }
) {
  return request<API.ApiResponseListVideoDecompositionBatchResponse>(
    "/api/video-script-decomposition/batches",
    {
      method: "GET",
      params: {
        ...params,
      },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/video-script-decomposition/batches */
export async function create(
  body: API.CreateVideoDecompositionBatchRequest,
  options?: { [key: string]: any }
) {
  return request<API.ApiResponseVideoDecompositionBatchResponse>(
    "/api/video-script-decomposition/batches",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/video-script-decomposition/batches/${param0} */
export async function detail(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.detailParams,
  options?: { [key: string]: any }
) {
  const { batchId: param0, ...queryParams } = params;
  return request<API.ApiResponseVideoDecompositionBatchResponse>(
    `/api/video-script-decomposition/batches/${param0}`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 GET /api/video-script-decomposition/episodes/${param0} */
export async function episode(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.episodeParams,
  options?: { [key: string]: any }
) {
  const { episodeId: param0, ...queryParams } = params;
  return request<API.ApiResponseVideoDecompositionEpisodeDetailResponse>(
    `/api/video-script-decomposition/episodes/${param0}`,
    {
      method: "GET",
      params: { ...queryParams },
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/video-script-decomposition/episodes/${param0}/confirm */
export async function confirm(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.confirmParams,
  body: API.ConfirmVideoDecompositionDraftRequest,
  options?: { [key: string]: any }
) {
  const { episodeId: param0, ...queryParams } = params;
  return request<API.ApiResponseVideoDecompositionEpisodeResponse>(
    `/api/video-script-decomposition/episodes/${param0}/confirm`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 PUT /api/video-script-decomposition/episodes/${param0}/draft */
export async function updateDraft(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.updateDraftParams,
  body: API.UpdateVideoDecompositionDraftRequest,
  options?: { [key: string]: any }
) {
  const { episodeId: param0, ...queryParams } = params;
  return request<API.ApiResponseVideoDecompositionEpisodeResponse>(
    `/api/video-script-decomposition/episodes/${param0}/draft`,
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/video-script-decomposition/episodes/${param0}/retry */
export async function retry(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.retryParams,
  body: API.RetryVideoDecompositionEpisodeRequest,
  options?: { [key: string]: any }
) {
  const { episodeId: param0, ...queryParams } = params;
  return request<API.ApiResponseVideoDecompositionEpisodeResponse>(
    `/api/video-script-decomposition/episodes/${param0}/retry`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      params: { ...queryParams },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /api/video-script-decomposition/uploads */
export async function upload(body: {}, options?: { [key: string]: any }) {
  return request<API.ApiResponseVideoDecompositionUploadResponse>(
    "/api/video-script-decomposition/uploads",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: body,
      ...(options || {}),
    }
  );
}
